parastell.py

```diff
@@ -93,36 +95,94 @@ def cubit_export(components, export, magnets):
+
+        # assign components to blocks
+        for comp in components.values():
+            cubit.cmd('set duplicate block elements off')
+            cubit.cmd("block " + str(comp['vol_id']) + " add volume " + str(comp['vol_id']))
+        
+        # assign materials to blocks
+        for comp in components.values():
+            cubit.cmd("block " + str(comp['vol_id']) + " material " + ''.join(("\'",comp['h5m_tag'],"\'")))
+            
+
+        if magnets is not None: #conditionally assign material to magnets
+            
+            magnet_h5m_tag = magnets['h5m_tag']
+            
+            # create magnet material
+            cubit.cmd(f'create material "{magnet_h5m_tag}" property_group "CUBIT-ABAQUS')
+
+            # assign magnets to block
+            block_number = min(magnets['vol_id'])
+            for vol in magnets['vol_id']:
+                cubit.cmd('set duplicate block elements off')
+                cubit.cmd("block " + str(block_number) + " add volume " + str(vol))
+            
+            # assign magnet material to block
+            cubit.cmd("block " + str(block_number) + " material " + ''.join(("\'",magnet_h5m_tag,"\'")))
+        
+        #mesh the model
+        cubit.cmd("set trimesher coarse on ratio " + str(anisotropic_ratio) + " angle " + str(deviation_angle))
+        cubit.cmd("surface all scheme trimesh")
+        cubit.cmd("mesh surface all")
+
+        #export dagmc file
+        cubit.cmd(f'export cf_dagmc "{cwd + "/dagmc.h5m"}" overwrite')
+
+    else:
+
+        # Extract Cubit export parameters
+        facet_tol = export['facet_tol']
+        len_tol = export['len_tol']
+        norm_tol = export['norm_tol']
+        # Conditionally assign magnet material group
+        if magnets is not None:
+            magnet_h5m_tag = magnets['h5m_tag']
+            cubit.cmd(
+                f'group "mat:{magnet_h5m_tag}" add volume '
+                + " ".join(str(i) for i in magnets['vol_id'])
+            )
+        
+        # Assign material groups
+        for comp in components.values():
+            cubit.cmd(f'group "mat:{comp["h5m_tag"]}" add volume {comp["vol_id"]}')
+
+        # Initialize tolerance strings for export statement as empty strings
+        facet_tol_str = ''
+        len_tol_str = ''
+        norm_tol_str = ''
+
+        # Conditionally fill tolerance strings
+        if facet_tol is not None:
+            facet_tol_str = f'faceting_tolerance {facet_tol}'
+        if len_tol is not None:
+            len_tol_str = f'length_tolerance {len_tol}'
+        if norm_tol is not None:
+            norm_tol_str = f'normal_tolerance {norm_tol}'
+        
+        # DAGMC export
         cubit.cmd(
-            f'group "mat:{magnet_h5m_tag}" add volume '
-            + " ".join(str(i) for i in magnets['vol_id'])
+            f'export dagmc "dagmc.h5m" {facet_tol_str} {len_tol_str} {norm_tol_str}'
+            f' make_watertight'
```

Comment:
It might make sense to put this in a function

Reply:
its now in the local function `legacy_export()`, though I am a bit unclear about which part specifically you meant to be in a function